Update SDK docs for Hosted Bundle (#63069)
metabase/metabase: docs/embedding/sdk/upgrade.md

Here's a basic overview of the steps you'll want to take when upgrading your SDK.

## 1. Read the release post and changelog for Metabase and the Embedded analytics SDK

- [Release posts](https://www.metabase.com/releases) give a good overview of what's in each release, and call out breaking changes (which are rare).
- [Metabase changelogs](https://www.metabase.com/changelog) list Metabase changes.
- [Embedded analytics SDK changelogs](https://www.metabase.com/changelog/56) list changes specific to the SDK Package.

> Check for any relevant changes, especially deprecation changes that require you to update your application's code. If there are deprecation changes, we'll have docs that'll walk you through what changes you'll need to make and why.

Reviewer: Why deprecation vs breaking? Are we claiming there will never be breaking changes?

Reviewer 2: I would put "deprecations or breaking changes" to cover both grounds; there will still be breaking changes but it will be much rarer than before.

Author: Starting from 58 with Hosted Bundle we should not have breaking changes between 2 sibling versions in a public API. We deprecate something first, so consumers will have 2 months to update their code. So technically it's not a breaking change.

Be sure to deploy your application changes and upgrade your Metabase in parallel so that the SDK version and the Metabase version stay in sync.

> ### Cached SDK Bundle
> 
> After upgrading, the previous version of the SDK Bundle may still be served from your new Metabase instance for up to 60 seconds (`Cache-Control: public, max-age=60`).
> This short cache window is intentional and helps ensure fast performance while still allowing updates to propagate quickly.
> 
> If you don’t see your changes immediately, clear browser cache or just wait a minute. After that, the SDK Package will begin loading the newly deployed SDK Bundle.

Reviewer: More descriptive heading, and active voice.

```suggestion
### Caching may delay the upgrade by up to a minute

This is intentional. After upgrading, Metabase may still serve the previous, cached version of the SDK Bundle for up to 60 seconds (`Cache-Control: public, max-age=60`). This short cache window helps ensure fast performance while still allowing updates to propagate quickly.

If you don’t see your changes immediately, clear your browser's cache or just wait a minute. After that, the SDK Package will load the newly deployed SDK Bundle.
```
